src/transformStateWithClones.js

```diff
@@ -7,7 +7,52 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesHistory = [];
+  let newState = state;
+
+  for (const action of actions) {
+    newState = { ...newState };
```

Comment:
[NAMING] - use proper names for object copy. Instead of `newState`, a better name would be `stateCopy` to clearly indicate that this variable is a copy of the state.